implementations/go/base/static/search/search_param_types.go

```diff
@@ -0,0 +1,705 @@
+package search
+
+import (
+	"fmt"
+	"math/big"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	//"github.com/davecgh/go-spew/spew"
+)
+
+// Query describes a string-based FHIR query and the resource it is associated
+// with.  For example, the URL http://acme.com/Condition?patient=123&onset=2012
+// should be represented as:
+// 	Query { Resource: "Condition", Query: "patient=123&onset=2012" }
+type Query struct {
+	Resource string
+	Query    string
+}
+
+// Params parses the query string and returns a slice containing the
+// appropriate SearchParam instances.  For example, a Query on the "Condition"
+// resource with the query string "patient=123&onset=2012" should return a
+// slice containing a ReferenceParam (for patient) and a DateParam (for onset).
+func (q *Query) Params() []SearchParam {
```

Comment:
I guessing that the panics here will result in a 500 response code. 501 may be more appropriate, but from reading the spec, I doesn't seem like there is a really good code to describe the situation: we understand the request, but we don't implement it yet.